desc/magnetic_fields.py

```diff
@@ -1803,6 +1810,299 @@ def from_surface(
+            UserWarning,
+            "Typical helicity (M,N) has N=NFP.",
+        )
+
+    def change_resolution(
+        self,
+        L_well=None,
+        M_well=None,
+        L_shift=None,
+        M_shift=None,
+        N_shift=None,
+        NFP=None,
+    ):
+        """Set the spectral resolution of well and shift terms.
+
+        Parameters
+        ----------
+        L_well : int
+            Radial resolution of the magnetic well parameters B_lm.
+        M_well : int
+            Number of spline points in the magnetic well parameters B_lm.
+        L_shift : int
+            Radial resolution of x_lmn.
+        M_shift : int
+            Poloidal resolution of x_lmn.
+        N_shift : int
+            Toroidal resolution of x_lmn.
+        NFP : int
+            Number of field periods.
+
+        """
+        old_L_well = self.L_well
+
+        self._NFP = setdefault(NFP, self.NFP)
+        self._L_well = setdefault(L_well, self.L_well)
+        self._M_well = setdefault(M_well, self.M_well)
+        self._L_shift = setdefault(L_shift, self.L_shift)
+        self._M_shift = setdefault(M_shift, self.M_shift)
+        self._N_shift = setdefault(N_shift, self.N_shift)
+
+        # change well parameters and basis
+        rho = (  # Chebyshev-Gauss-Lobatto nodes
+            1
+            - np.cos(np.arange(old_L_well // 2, old_L_well + 1, 1) * np.pi / old_L_well)
+        ) / 2
+        nodes = np.array([rho, np.zeros_like(rho), np.zeros_like(rho)]).T
+
+        transform_fwd = self.well_basis.evaluate(nodes)
+        transform_rev = scipy.linalg.pinv(transform_fwd)
+        B_old = transform_fwd @ self.B_lm.reshape((old_L_well + 1, -1))
+
+        eta_old = np.linspace(0, jnp.pi / 2, num=B_old.shape[-1])
+        eta_new = np.linspace(0, jnp.pi / 2, num=self.M_well)
+
+        B_new = np.zeros((old_L_well + 1, self.M_well))
+        for i in range(old_L_well + 1):
+            B_new[i, :] = interp1d(eta_new, eta_old, B_old[i, :], method="monotonic-0")
+        B_lm_old = transform_rev @ B_new
+
+        old_modes_well = self.well_basis.modes
+        self.well_basis.change_resolution(self.L_well)
+
+        B_lm_new = np.zeros((self.L_well + 1, self.M_well))
+        for j in range(self.M_well):
+            B_lm_new[:, j] = copy_coeffs(
+                B_lm_old[:, j], old_modes_well, self.well_basis.modes
+            )
+        self._B_lm = B_lm_new.flatten()
+
+        # change shift parameters and basis
+        old_modes_shift = self.shift_basis.modes
+        self.shift_basis.change_resolution(
+            self.L_shift, self.M_shift, self.N_shift, NFP=self.NFP, sym="cos(t)"
+        )
+        self._x_lmn = copy_coeffs(self.x_lmn, old_modes_shift, self.shift_basis.modes)
+
+    def compute(
+        self,
+        names,
+        grid=None,
+        params=None,
+        transforms=None,
+        profiles=None,
+        data=None,
+        **kwargs,
```

Comment:
a new `kwarg`, `iota` was added in this PR. Just a general thought, we should have some documentation of what compute functions need what kwargs, maybe the tutorials are enough but I still think some documentation would be good

Reply:
This is noted by Issue #707. I agree but think we can add this in a future PR.